man/fi_mr.3.md

```diff
@@ -546,6 +552,22 @@ region.  The domain authorization key will be used if the auth_key_size
 provided is 0.  This field is ignored unless the fabric is opened with API 
 version 1.5 or greater.
 
+## iface
+Indicates the software interfaces used by the application to allocate and
+manage the memory region:
+
+*FI_HMEM_SYSTEM*
+: Uses standard operating system calls and libraries, such as malloc,
+  calloc, realloc, mmap, and free.
+
+*FI_HMEM_CUDA*
+: Uses Nvidia CUDA interfaces.
+
+## pdevice_id 
+
+Pointer to the device identifier. This field is ignored if iface is
+FI_HMEM_SYSTEM.
```

Comment:
Gurgle, CUDU uses the typedef "CUdeviceptr".  So the cuda name can be confusing when paired with this name.  CUDA defines CUdeviceptr as "an unsigned integer type whose size matches the size of a pointer".  So, maybe we switch to this definition:
```
device_id: The device identifier.  For FI_HMEM_CUDA, this value should be set to a CUdeviceptr.  For FI_HMEM_SYSTEM, this field is ignored.
```
This is a change from having the field be the address of the device identifier to being the identifier itself, but I think will end up being clearer than having our device pointer be a pointer to a cude device pointer...

Reply:
I thought this field was going to be a handle to the device itself. In the CUDA case, this would be something we would pass to `cudaSetDevice()` for instance, in the presence of multiple GPUs. CUdeviceptr would just be a pointer to the CUDA buffer right?

Reply:
Also, I am neither a CUDA or a CUDU expert, so correct me if I am thinking about this completely wrong.

Reply:
Gah - yes, I think you're right.  CUdeviceptr is a pointer to... memory on the device?  What a horrible name.  I think we're wanting CUdevice here.  That's a device, right?  That's also just an int, I think, since cudaSetDevice / GetDevice use ints.

So, keep this as is and define pdevice_id as... an int * when used with CUDA?

Now I wonder if we shouldn't make this an unnamed union, so we can match exactly based on the iface value:
'''
union {
    uint64_t reserved;
    int cuda_device;
};
'''

Reply:
A union would be nice 👍